test/python/result/test_sampled_expval.py

```diff
@@ -85,6 +86,14 @@ def test_same(self):
         exp3 = sampled_expectation_value(counts, spo)
         self.assertAlmostEqual(exp3, ans)
 
+        so = SparseObservable.from_label(oper)
+        exp4 = sampled_expectation_value(counts, so)
+        self.assertAlmostEqual(exp4, ans)
+
+        so_non_diag = SparseObservable.from_label(oper_non_diag)
+        with self.assertRaisesRegex(ValueError, "Operator string .* contains non-diagonal terms"):
+            sampled_expectation_value(counts, so_non_diag)
```

Comment:
Since this has a return value it's nice to show we're deliberately not using it:
```suggestion
            _ = sampled_expectation_value(counts, so_non_diag)
```